addons/website_slides_survey/tests/test_course_certification_stats.py

```diff
@@ -0,0 +1,97 @@
+
+class TestCourseCertificationStats(TransactionCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        # Create courses and certifications
+        cls.channels = cls.env['slide.channel'].create([
+            {
+                'name': 'Course 1',
+                'enroll': 'public',
+                'is_published': True,
+            },
+            {
+                'name': 'Course 2',
+                'enroll': 'public',
+                'is_published': True,
+            },
+        ])
+
+        cls.certifications = cls.env['survey.survey'].create([
+            {
+                'title': 'certification 1',
+                'scoring_type': 'scoring_with_answers',
+                'certification': True,
+            },
+            {
+                'title': 'certification 2',
+                'scoring_type': 'scoring_with_answers',
+                'certification': True,
+            }]
+        )
+
+        # Link the certifications to courses
+        cls.slide_certifications = cls.env['slide.slide'].create([
+            {
+                'name': 'Certification 1 slide',
+                'channel_id': cls.channels.ids[0],
+                'slide_category': 'certification',
+                'survey_id': cls.certifications.ids[0],
+                'is_published': True,
+            },
+            {
+                'name': 'Certification 2 slide',
+                'channel_id': cls.channels.ids[1],
+                'slide_category': 'certification',
+                'survey_id': cls.certifications.ids[1],
+                'is_published': True,
+            },
+        ])
+
+        # Create course participants
+        cls.portal_users = [
+            mail_new_test_user(
+                cls.env, login=f'portal_test {i + 1}',
+                groups='base.group_portal',
+                name=f'Portal {i + 1}')
+            for i in range(3)
+        ]
+
+        cls.channel_partners = cls.env['slide.channel.partner'].create([
+            {
+                'channel_id': channel_id,
+                'partner_id': user.partner_id.id,
+            }
+            for channel_id in cls.channels.ids for user in cls.portal_users
+        ])
+
+        # Set certified/not certified participants
+        cls.slide_partners = cls.env['slide.slide.partner'].create([
+            {
+                'channel_id': channel_id.id,
+                'partner_id': user.partner_id.id,
+                'slide_id': channel_id.slide_ids.ids[0],
+            }
+            for channel_id in cls.channels for user in cls.portal_users
+        ])
+
+        cls.slide_partners[0].survey_scoring_success = True
+
+    def test_channel_members_certified_count(self):
+        self.assertEqual(self.channels[0].members_certified_count, 1, 'Course 1 has 1 certified participant')
+        self.assertEqual(self.channels[1].members_certified_count, 0, 'Course 2 has 0 certified participant')
```

Comment:
Could you here please get `members_certified_counts` for both courses in one call, then check the values? This way the compute is only called once and we test that it correctly assigns values for multiple records in batch. Thanks